decompiler/IR2/Form.cpp

```diff
@@ -740,6 +794,69 @@ void EmptyElement::apply_form(const std::function<void(Form*)>&) {}
 void EmptyElement::collect_vars(RegAccessSet&) const {}
 void EmptyElement::get_modified_regs(RegSet&) const {}
 
+/////////////////////////////
+// RLetElement
+/////////////////////////////
+
+bool cmp(Register x, Register y) {
+  int comparison = x.to_string().compare(y.to_string());
+  if (comparison <= 0)
+    return true;
+  return false;
+}
+
+RLetElement::RLetElement(Form* _body, RegSet _regs) : body(_body) {
+  for (auto& reg : _regs) {
+    sorted_regs.push_back(reg);
+  }
+  std::sort(sorted_regs.begin(), sorted_regs.end(), cmp);
+}
+
+void RLetElement::apply(const std::function<void(FormElement*)>& f) {
+  // note - this is done in program order, rather than print order. Not sure if this makes sense.
+  f(this);
+  body->apply(f);
+}
+
+goos::Object RLetElement::to_form_internal(const Env& env) const {
+  std::vector<goos::Object> regs;
+  for (auto& reg : sorted_regs) {
+    if (reg.get_kind() == Reg::RegisterKind::VF ||
+        reg.get_kind() == Reg::RegisterKind::COP2_MACRO_SPECIAL) {
+      std::string reg_name = reg.to_string() == "ACC" ? "acc" : reg.to_string();
+      regs.push_back(
+          pretty_print::build_list(pretty_print::to_symbol(fmt::format("{} :class vf", reg_name))));
+    }
+  }
+
+  std::vector<goos::Object> rletForm;
+  rletForm.push_back(pretty_print::to_symbol("rlet"));
+  rletForm.push_back(pretty_print::build_list(regs));
+
+  // NOTE - initialize any relevant registers in the body first
+  for (auto& reg : sorted_regs) {
+    if (reg.get_kind() == Reg::RegisterKind::VF && reg.to_string() == "vf0") {
+      rletForm.push_back(
+          pretty_print::to_symbol("(.lvf vf0 (new 'static 'vector :x 0.0 :y 0.0 :z 0.0 :w 1.0))"));
```

Comment:
Don't worry about it for this PR, but it might be a good idea to move this to a macro like `initialize-constant-vector!` or something.  There could be some clever way to do this initialization that's faster that a normal static load.